lib/LaTeXML/Package/LaTeX.pool.ltxml

```diff
@@ -4418,9 +4418,22 @@ our %index_style = (textbf => 'bold', bf => 'bold', textrm => '', rm => '',
 sub process_index_phrases {
   my ($gullet, $phrases, $inlist) = @_;
   my @expansion = ();
+  my @tokens    = $phrases->unlist;
+  # check we have a well-formed argument
+  return unless @tokens;
+  my $group_level = 0;
+  for my $t (@tokens) {
+    my $cc = $t->getCatcode;
+    if    ($cc == CC_BEGIN) { $group_level++; }
+    elsif ($cc == CC_END) {
+      $group_level--;
+      # discard if unbalanced close group;
+      last if ($group_level < 0); } }
+  if ($group_level != 0) {    # if a group is still open by the end, ill-formed, discard;
+    Warn("malformed", "indexentry", $gullet,
+      'index entry has unbalanced groups, discarding: "' . ToString($phrases) . '"');
+    return; }
```

Comment:
Have you seen `$tokens->isBalanced` ?

Reply:
Excellent reminder, and it even allowed me to improve that method a tiny bit.